src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateUtils.java

```diff
@@ -92,6 +93,7 @@ public static ContainerTemplate combine(@CheckForNull ContainerTemplate parent,
         String args = Strings.isNullOrEmpty(template.getArgs()) ? parent.getArgs() : template.getArgs();
         boolean ttyEnabled = template.isTtyEnabled() ? template.isTtyEnabled() : (parent.isTtyEnabled() ? parent.isTtyEnabled() : false);
         String resourceRequestCpu = Strings.isNullOrEmpty(template.getResourceRequestCpu()) ? parent.getResourceRequestCpu() : template.getResourceRequestCpu();
+        Long runAsUser = parent.getRunAsUser()  ==  Integer.toUnsignedLong(0) ? parent.getRunAsUser() : template.getRunAsUser();
```

Comment:
Do not use `==` to compare `Long` objects.

Reply:
fixed